### PR TITLE
add linters

### PR DESCRIPTION
**Pull request by katherinelc321**
This PR adds linter validation and unit testing to makefile https://issues.redhat.com/browse/OSD-8597

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,3 +43,14 @@ get-logs:
...
+GOLANGCI_LINT_CACHE ?= /tmp/golangci-cache
+
+.PHONY: lint
+lint: 
+	GOLANGCI_LINT_CACHE=${GOLANGCI_LINT_CACHE}
+	-@golangci-lint run 
+
+.PHONY: test
+test: 
+	@go test ./...
```
Comment (dastergon) [inline Makefile:56]: can you add a new extra line?

**Comment by dastergon** [comment]
@katherinelc321 can you add a description in your PR?

File: Makefile (hunk 1 above)
Comment (katherinelc321) [reply Makefile:56]: an extra line where?

**Pull request merged**